core/src/main/java/io/cucumber/core/exception/CompositeCucumberException.java

```diff
@@ -1,25 +1,32 @@
 package io.cucumber.core.exception;
 
+import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
+import java.util.Objects;
+
+public class CompositeCucumberException
+    extends CucumberException {
 
-public class CompositeCucumberException extends CucumberException {
     private final List<Throwable> causes;
 
-    public CompositeCucumberException(List<Throwable> causes) {
-        super(String.format("There were %d exceptions:", causes.size()));
-        this.causes = causes;
+    public CompositeCucumberException(final List<Throwable> causes) {
+        super(String.format("There were %d exceptions:",
+            Objects.isNull(causes) ? 0 : causes.size()
+        ));
+        this.causes = Objects.isNull(causes) ? new ArrayList<>() : causes;
```

Comment:
This exception shouldn't be used in scenarios where there is only one exception. Throwing an `IllegalArgument` exception when an empty is provided would be better.

It is also preferable to use `== null` for regular null checks. `Objects::isNull` is intended for use in lambda expressions.

Reply:
will do tomorrow